packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -90,14 +90,23 @@ class CodeSizeController {
   final _activeDiffTreeType =
       ValueNotifier<DiffTreeType>(DiffTreeType.combined);
 
-  // TODO(peterdjlee): Cache each diff tree so that it's less expensive
-  //                   to change bettween diff tree types.
+  TreemapNode _increasedDiffTreeRoot;
+  TreemapNode _decreasedDiffTreeRoot;
+  TreemapNode _combinedDiffTreeRoot;
```

Comment:
make these getters that either return the cached value or generate the tree if the cached value is null

Reply:
all the cached values also need to be reset to null on clear